qiime/plugin.py

```diff
@@ -77,26 +80,101 @@ def register_type_to_data_layout(self, semantic_type, name, version):
 
 # TODO refactor these classes, they are basically the same
 class PluginMethods(dict):
-    def __init__(self, package):
+    def __init__(self, plugin, package):
```

Comment:
Can you call this `plugin_name`, or pass the actual `Plugin` object? The current name is confusing